src/tengbe.py

```diff
@@ -275,6 +276,37 @@ def read_counters(self):
                 results[reg] = self.parent.memory_devices[reg].read()
         return results
 
+    def tx_okay(self,wait_time=1):
+        """
+        Is this gbe block okay?
+        :return: True/False
+        """
+        try:
+            result0 = self.read_tx_counters()
+            time.sleep(wait_time) # what number is best?
+            result1 = self.read_tx_counters()
+            # if debug register are implemented
+            # check if errors are not changing
+            assert (result0[self.name+'_txfullctr']['data']['reg'] ==
+                    result1[self.name+'_txfullctr']['data']['reg'])
+            assert (result0[self.name+'_txofctr']['data']['reg'] ==
+                    result1[self.name+'_txofctr']['data']['reg'])
+            assert (result0[self.name+'_txerrctr']['data']['reg'] ==
+                    result1[self.name+'_txerrctr']['data']['reg'])
+            # check increment
+            assert (result1[self.name+'_txctr']['data']['reg'] -
+                    result0[self.name+'_txctr']['data']['reg'] > 0)
+            assert (result1[self.name+'_txvldctr']['data']['reg'] -
+                    result0[self.name+'_txvldctr']['data']['reg'] > 0)
+        except KeyError:
+            raise RuntimeError('Gbe block {} does not have the necessary '
+                               'registers to check TX status.'.format(self.name))
+        except:
```

Comment:
It is 'dangerous' to have  a naked 'except', since that catches _all_ exceptions, including ones you don't normally want to catch such as 'SystemExit' and 'KeyboardInterrupt'. What you want here is:

except Exception: